sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JacksonParser.scala

```diff
@@ -325,6 +328,36 @@ class JacksonParser(
         parseJsonToken[AnyRef](parser, dataType)(PartialFunction.empty[JsonToken, AnyRef])
   }
 
+  def makeKeyConverter(dataType: DataType): KeyConverter = dataType match {
```

Comment:
Can we reuse `makeConverter`?

Reply:
I think it's different from `makeConverter`, as the JSON key is always string: `{"key": value}`.

That said, I'm not sure if this feature makes sense. If we turn a JSON string into a map, the key type should always be string type. Users can run `transform_keys` function to cast the key string to whatever data type they want.

We should probably improve the error message though, if a user specifies a MapType with non-string key type. cc @MaxGekk

Reply:
> Can we reuse `makeConverter`?

The procesing of the keys is different as Cloud-fan said so I think we cant reuse it.

Reply:
+1 for @cloud-fan's idea